cmake/Dependencies.cmake

```
@@ -1266,6 +1266,20 @@ endif()
 
 # ---[ HIP
 if(USE_ROCM)
+  # This prevents linking in the libtinfo from /opt/conda/lib which conflicts with ROCm libtinfo.
+  # Currently only active for Ubuntu 20.04 and greater versions.
+  if(UNIX)
+    execute_process(COMMAND bash -c "cat /etc/os-release | grep -e '^NAME=' | awk -F\\\" '{print $2}'" OUTPUT_VARIABLE OS_DISTRO OUTPUT_STRIP_TRAILING_WHITESPACE)
```

It would be nice to use CMake primitives, such as `file` and `REGEX REPLACE`
Not tested, but something along the following lines would probably do
```suggestion
    file(STRINGS /etc/os-release OS_RELEASE_TEXT)
    string(REGEX REPLACE "^NAME=(\w+)*" "\\1" OBJCOPY_VERSION_MAJOR ${OS_DISTRO})
```

---

Addressed in https://github.com/pytorch/pytorch/pull/83552/commits/956ade57e16f893a556408620949a24931510958